cmd/memtierd/main.go

```diff
@@ -0,0 +1,331 @@
+	// This is the container (post-)stop request handler. You can update any
+	// of the remaining running containers. Take a look at the functions in
+	// pkg/api/update.go to see the available controls.
+	//
+
+	podName := pod.GetName()
+	dirPath := fmt.Sprintf("/tmp/memtierd/%s", podName)
+
+	// Kill the memtierd process
+	out, err := exec.Command("sudo", "pkill", "-f", dirPath).CombinedOutput()
+	if err != nil {
+		exitErr, ok := err.(*exec.ExitError)
+		if !ok || exitErr.ExitCode() != 1 {
+			// Error occurred that is not related to "no processes found"
+			log.Fatalf("Error killing memtierd process: %v. Output: %s\n", err, out)
+		} else {
+			// "No processes found" error, do nothing
+			log.Printf("No processes found for memtierd process\n")
+		}
+	}
+
+	err = os.RemoveAll(dirPath)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	return []*api.ContainerUpdate{}, nil
+}
+
+func (p *plugin) onClose() {
+	log.Infof("Connection to the runtime lost, exiting...")
+	os.Exit(0)
+}
+
+func getFullCgroupPath(ctr *api.Container) string {
+	cgroupPath := ctr.Linux.CgroupsPath
+
+	split := strings.Split(cgroupPath, ":")
+
+	partOne := split[0]
+	partTwo := fmt.Sprintf("%s-%s.scope", split[1], split[2])
+
+	partialPath := fmt.Sprintf("%s/%s", partOne, partTwo)
+
+	fullPath := fmt.Sprintf("*/kubepods*/%s", partialPath)
+
+	file, err := os.Open("/proc/mounts")
+	if err != nil {
+		log.Fatalf("failed to open /proc/mounts: %v", err)
+	}
+	defer file.Close()
+
+	cgroupMountPoint := ""
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		fields := strings.Fields(line)
+
+		if fields[0] == "cgroup2" {
+			cgroupMountPoint = fields[1]
+			break
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		log.Fatalf("failed to read /proc/mounts: %v", err)
+	}
+
+	// Find the cgroup path corresponding to the container
+	var fullCgroupPath string
+	err = filepath.WalkDir(cgroupMountPoint, func(path string, info os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			matches, err := filepath.Glob(filepath.Join(path, fullPath))
+			if err != nil {
+				return err
+			}
+
+			if len(matches) > 0 {
+				fullCgroupPath = matches[0]
+				return filepath.SkipDir
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		log.Fatalf("failed to traverse cgroup directories: %v", err)
+	}
+
+	if fullCgroupPath == "" {
+		log.Fatalf("cgroup path not found")
+	}
+
+	log.Printf("Cgroup path: %s", fullCgroupPath)
+
+	return fullCgroupPath
+}
+
+func editMemtierdConfig(fullCgroupPath string, podName string, containerName string, template string) (string, string, string) {
+	templatePath := fmt.Sprintf("/templates/%s", template)
+	yamlFile, err := ioutil.ReadFile(templatePath)
+	if err != nil {
+		log.Fatalf("Error reading YAML file: %v\n", err)
+	}
+
+	// Create pod directory if it doesn't exist
+	podDirectory := fmt.Sprintf("/tmp/memtierd/%s", podName)
+	if err := os.MkdirAll("/host"+podDirectory, 0755); err != nil {
```

Comment:
The `/host` part must not be hard coded but should be given as a parameter to this program.